python/src/transaction.rs

```diff
@@ -0,0 +1,249 @@
+                Ok(Self(op))
+            }
+            "Restore" => {
+                let version = ob.getattr("version")?.extract()?;
+                let op = Operation::Restore { version };
+                Ok(Self(op))
+            }
+            "Rewrite" => {
+                let groups = extract_vec(&ob.getattr("groups")?)?;
+                let rewritten_indices = extract_vec(&ob.getattr("rewritten_indices")?)?;
+                let op = Operation::Rewrite {
+                    groups,
+                    rewritten_indices,
+                };
+                Ok(Self(op))
+            }
+            "CreateIndex" => {
+                let uuid = ob.getattr("uuid")?.extract()?;
+                let name = ob.getattr("name")?.extract()?;
+                let fields = ob.getattr("fields")?.extract()?;
+                let dataset_version = ob.getattr("dataset_version")?.extract()?;
+
+                let fragment_ids = ob.getattr("fragment_ids")?;
+                let fragment_ids_ref: &Bound<'_, PySet> = fragment_ids.downcast()?;
+                let fragment_ids = fragment_ids_ref
+                    .into_iter()
+                    .map(|id| id.extract())
+                    .collect::<PyResult<Vec<u32>>>()?;
+                let fragment_bitmap = Some(fragment_ids.into_iter().collect());
+
+                let new_indices = vec![Index {
+                    uuid: Uuid::parse_str(uuid)
+                        .map_err(|e| PyValueError::new_err(e.to_string()))?,
+                    name,
+                    fields,
+                    dataset_version,
+                    fragment_bitmap,
+                    // TODO: we should use lance::dataset::Dataset::commit_existing_index once
+                    // we have a way to determine index details from an existing index.
+                    index_details: None,
+                }];
+
+                let op = Operation::CreateIndex {
+                    removed_indices: Vec::new(),
+                    new_indices,
+                };
+                Ok(Self(op))
+            }
+            unsupported => Err(PyValueError::new_err(format!(
+                "Unsupported operation: {unsupported}",
+            ))),
+        }
+    }
+}
+
+impl ToPyObject for PyLance<&Operation> {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let namespace = py
+            .import_bound(intern!(py, "lance"))
+            .and_then(|module| module.getattr(intern!(py, "LanceOperation")))
+            .expect("Failed to import LanceOperation namespace");
+
+        match self.0 {
+            Operation::Append { ref fragments } => {
+                let fragments = export_vec(py, fragments.as_slice());
+                let cls = namespace
+                    .getattr("Append")
+                    .expect("Failed to get Append class");
+                cls.call1((fragments,)).unwrap().to_object(py)
+            }
+            _ => todo!(),
```

Comment:
Do we actually use `ToPyObject` for `Append` or is this more just an example of how you might do this in case we need to later.

Reply:
`Dataset::commit_batch` returns a `PyLance<Transaction>`, which will contain an operation. Currently just Append.